stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -3269,7 +3252,7 @@ TreeNode* YangParseTreePaths::AddSubtreeInterface(
 
   node = tree->AddNode(GetPath("interfaces")(
       "interface", name)("state")("hardware-port")());
-  SetUpInterfacesInterfaceStateHardwarePort(node_id, port_id, node, tree);
+  SetUpInterfacesInterfaceStateHardwarePort(node, tree);
```

Comment:
we can also pass the `name` to the function so we don't need to parse the node to get the name.

Reply:
yeah, the whole reference leaf thing is not really needed here as the name is exactly the same.
I think I'll do what you suggest, but keep this code change for later reference.